src/js_app.hpp

```diff
@@ -355,5 +382,46 @@ void AppClass<T>::set_versions(ContextType ctx, ObjectType this_object, Argument
     AppClass<T>::platform_version = Object::validated_get_string(ctx, versions, "platformVersion");
 }
 
+template <typename T>
+void AppClass<T>::add_listener(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue& return_value)
+{
+    auto callback = Value::validated_to_function(ctx, args[0]);
```

Comment:
```suggestion
    auto callback = Value::validated_to_function(ctx, args[0], "callback");
```

If you pass a string as the third argument to `validated_to_xxx` then you get a more descriptive exception: https://github.com/realm/realm-js/blob/master/src/js_types.hpp#L213 and https://github.com/realm/realm-js/blob/master/src/js_types.hpp#L121